frontend/taipy/src/ScenarioSelector.tsx

```diff
@@ -466,6 +444,33 @@ const ScenarioSelector = (props: ScenarioSelectorProps) => {
         ]
     );
 
+    const onDialogOpen = useCallback(() => {
+        if (showDialog) {
+            setOpen(true);
+            setActionEdit(false);
+        } else {
+            onSubmit(false, false, {}, false);
+        }
+    }, [showDialog, onSubmit]);
+
+    const onDialogClose = useCallback(() => {
```

Comment:
Does this need to be a hook?

Reply:
yes because it depends on setState actions that are only available in the component